tests/e2e/tests/mixer/mixer_test.go

```diff
@@ -344,33 +366,44 @@ func TestRateLimit(t *testing.T) {
 
 	got, err := vectorValue(value, map[string]string{responseCodeLabel: "429"})
 	if err != nil {
-		t.Errorf("Could not find rate limit value: %v", err)
+		t.Logf("prometheus values for request_count:\n%s", promDump(promAPI, "request_count"))
+		t.Fatalf("Could not find rate limit value: %v", err)
 	}
 
-	// establish some baseline (40% of expected counts)
-	want := math.Floor(perSvc * .4)
+	// establish some baseline to protect against flakiness due to randomness in routing
+	want := math.Floor(want429s * .75)
 
-	// check rejections
+	// check resource exhausteds
 	if got < want {
 		t.Errorf("Bad metric value for rate-limited requests (429s): got %f, want at least %f", got, want)
 	}
 
 	got, err = vectorValue(value, map[string]string{responseCodeLabel: "200"})
 	if err != nil {
-		t.Log(promDump(promAPI, "request_count"))
-		t.Errorf("Could not find successes value: %v", err)
+		t.Logf("prometheus values for request_count:\n%s", promDump(promAPI, "request_count"))
+		t.Fatalf("Could not find successes value: %v", err)
 	}
 
-	succWant := opts.Duration.Seconds() * 4 // really want 5 qps, but settle for 4
-	want = math.Min(want, succWant)         // use the smaller of the two to allow for traffic variations
+	// establish some baseline to protect against flakiness due to randomness in routing
+	want = math.Floor(want200s * .75)
 
 	// check successes
 	if got < want {
-		t.Log(promDump(promAPI, "request_count"))
+		t.Logf("prometheus values for request_count:\n%s", promDump(promAPI, "request_count"))
 		t.Errorf("Bad metric value for successful requests (200s): got %f, want at least %f", got, want)
 	}
 }
 
+func allowRuleSync() {
+	glog.Info("Sleeping to allow rules to take effect...")
+	time.Sleep(1 * time.Minute)
+}
+
+func allowPrometheusSync() {
+	glog.Info("Sleeping to allow prometheus to record metrics...")
```

Comment:
:+1:
(we could also have our own sleep that says how long it's about to sleep with optional "why" message)